scripts/proto/README.md

```diff
@@ -1,166 +1,186 @@
-or have feature suggestions.
+This is still a work in progress, so please open an issue if you find
+any problems or have feature suggestions.
 
 # Setup
 
-* This requires Python 3. You may need to load the relevant module for your
-environment to be set up right.
-* You will need the Python protobuf module, which can be locally installed with
-`pip3 install --user protobuf`.
-* You will need a build of LBANN.
-
-_Advanced users_: This requires the `lbann_pb2` Python module generated from
-`lbann.proto` using the `protoc` compiler. The LBANN build process should do
-this automatically, installing it to `LBANN_BUILD_DIR/install/share/python`.
-This package uses some basic heuristics to attempt to locate this if
-`lbann_pb2` is not in your default Python search path. If these fail, you
-should manually set your Python path to include the directory with `lbann_pb2`
-(e.g. by setting the `PYTHONPATH` environment variable).
+Requirements:
+* Python 3. You may need to load the relevant module for your
+  environment to be set up right.
+* The Python protobuf module, which can be locally installed with
+  `pip3 install --user protobuf`.
+* A build of LBANN.
+
+_Advanced users_: This requires the `lbann_pb2` Python module
+generated from `lbann.proto` using the `protoc` compiler. The LBANN
+build process should do this automatically, installing it to
+`LBANN_BUILD_DIR/install/share/python`.  This package uses some basic
+heuristics to attempt to locate this if `lbann_pb2` is not in your
+default Python search path. If these fail, you should manually set
+your Python path to include the directory with `lbann_pb2` (e.g. by
+setting the `PYTHONPATH` environment variable).
 
 # Use
 
-This consists of two components, `lbann_proto` and `blocks`. `lbann_proto` is an
-automatically generated interface to (most of) the components of the LBANN
-prototext system, including all layers, callbacks, objective functions,
-regularizers, and metrics. `blocks` consists of manually-curated higher-level
+This consists of two components, `lbann_proto` and
+`lbann_modules`. `lbann_proto` is an automatically generated interface
+to (most of) the components of the LBANN prototext system, e.g. the
+layers, weights, objective functions, metrics, and
+callbacks. `lbann_modules` consists of manually-curated higher-level
 building blocks that are commonly used in neural networks.
 
+_A possible point of confusion_: LBANN uses the term "layer" in a
+somewhat unpopular manner. In LBANN and Caffe, a layer is a single
+operation that is applied to a fixed set of input tensors to obtain a
+fixed set of output tensors. In PyTorch, TensorFlow, and Keras, a
+layer is a pattern of operations that can be applied multiple times to
+different sets of input tensors, each time obtaining a different set
+of output tensors. Their usage of the word "layer" is closer to our
+usage of the word "module." In fact, PyTorch's "layers" are subclasses
+of their `Module` class.
```

Comment:
I would clarify that part of this is because, in this Python interface, we're not really doing forward/back prop, but just setting up the structure of the network.